packages/solid/src/create_query_resource.ts

```diff
@@ -11,18 +16,13 @@ function createQueryResource<Params, Data, Error>(
     equals: false,
   });
 
-  const { data, error, pending, start } = useUnit({
-    data: query.$data,
-    error: query.$error,
-    pending: query.$pending,
-    start: query.start,
-  });
+  const { data, error, pending, start } = useUnit(query);
 
   let dataDefer = createDefer<Data, Error>();
 
-  createEffect(() => {
+  createComputed(() => {
```

Comment:
Sorry, I did not get it. Why did you replace `createEffect` by `createComuted`? Can we have some tests for this change as well?

Reply:
`createComputed` starts computations immediately, not waiting for render.

Seems more correct in this situation.

Reply:
Okay. However, it would be nice to add test that would break with createEffect but not with createComputed